sympy/physics/control/routh_table.py

```diff
@@ -0,0 +1,189 @@
+        >>> RouthHurwitz(p, s).first_column
+        Matrix([
+        [               b_1],
+        [               b_2],
+        [-b_1*b_4/b_2 + b_3],
+        [               b_4]])
+
+        Here you can see how the table appears in the first column zero case:
+
+        >>> p1 = s**4 + s**3 + 3*s**2 + 3*s + 3
+        >>> RouthHurwitz(p1, s)
+        Matrix([
+        [            1, 3, 3],
+        [            1, 3, 0],
+        [      epsilon, 3, 0],
+        [3 - 3/epsilon, 0, 0],
+        [            3, 0, 0]])
+        >>> RouthHurwitz(p1, s).zero_rows_case
+        False
+
+        Here you can see how the table appears in the full row zero case (poles with only imaginary part):
+
+        >>> p2 = s**6 + 2*s**5 + 8*s**4 + 12*s**3 + 20*s**2 + 16*s + 16
+        >>> RouthHurwitz(p2, s)
+        Matrix([
+        [  1,  8, 20, 16],
+        [  2, 12, 16,  0],
+        [  2, 12, 16,  0],
+        [  8, 24,  0,  0],
+        [  6, 16,  0,  0],
+        [8/3,  0,  0,  0],
+        [ 16,  0,  0,  0]])
+        >>> RouthHurwitz(p2, s).zero_rows_case
+        True
+        >>> RouthHurwitz(p2, s).auxiliary_polynomial
+        Poly(2*s**4 + 12*s**2 + 16, s, domain='ZZ')
+
+        References
+        ==========
+        .. [1] https://en.wikipedia.org/wiki/Routh-Hurwitz_stability_criterion
+
+    """
+    def __new__(cls, polynomial, var, infinitesimal_element = None):
+        if not isinstance(var, Symbol):
+            raise ValueError("var must be a Symbol")
+        n = Poly(polynomial, var).degree()
+
+        return super().__new__(cls, n + 1, n//2 + 1, [0]*(n + 1)*(n//2 + 1))
+
+    def __init__(self, polynomial, var, infinitesimal_element = None):
+        self._var = var
+        self._polynomial = Poly(polynomial, var)
+        self._poly_degree = self._polynomial.degree()
+        self._coeffs = self._polynomial.all_coeffs()
+
+        self._zero_row_case = False
+        self._aux_poly_degree = 0
+
+        self._inf_element = infinitesimal_element
+        if infinitesimal_element is None:
+            self._inf_element = symbols("epsilon", dummy=True)
+
+        self._build_table()
+
+    def _build_table(self):
+        self._initialize()
+        self._calculate()
+
+    def _initialize(self):
+        row, col = 0, 0
+        for coeff in self._coeffs:
+            self[row, col] = coeff
+
+            row = (row+1) % 2
+            col = col + 1 - row
+
+        if self[1, 0] != 0:
+            return
+
+        self._handle_special_cases(1)
+
+    def _calculate(self):
+        for i in range(2, self.rows):
+            self._calculate_row(i)
+            self._handle_special_cases(i)
+
+    def _calculate_row(self, i):
+        trailing_columns = i//2
+        for j in range(self.cols - trailing_columns):
+            num = (self[i-1, 0] * self[i-2, j+1]
+                   - self[i-2, 0] * self[i-1, j+1])
+            den = self[i-1, 0]
+
+            self[i, j] = simplify(num / den)
```

Comment:
We do our best to avoid calling simplify() in library code. Either don't simplify (as it is usually unecessary) or apply a specific targeted simplification routine if you absolutely have to.